options.go

```diff
@@ -33,19 +29,93 @@ type options struct {
 	// the cooldown time after every type of dump
 	// interval for cooldown，default 1m
 	// the cpu/mem/goroutine have different cooldowns of their own
+
+	// todo should we move CoolDown into Gr/CPU/MEM/GCheap Opts and support
+	// set different `CoolDown` for different opts?
 	CoolDown time.Duration
 
 	// if current cpu usage percent is greater than CPUMaxPercent,
 	// holmes would not dump all types profile, cuz this
 	// move may result of the system crash.
 	CPUMaxPercent int
 
-	logOpts    *loggerOptions
-	GrOpts     *grOptions
-	MemOpts    *memOptions
-	GCHeapOpts *gcHeapOptions
-	CPUOpts    *cpuOptions
-	ThreadOpts *threadOptions
+	logOpts *loggerOptions
+	grOpts  *grOptions
+
+	memOpts    *typeOption
+	gCHeapOpts *typeOption
+	cpuOpts    *typeOption
+	threadOpts *typeOption
+}
+
+// DumpOptions contains configuration about dump file.
+type DumpOptions struct {
+	// full path to put the profile files, default /tmp
+	DumpPath string
+	// default dump to binary profile, set to true if you want a text profile
+	DumpProfileType dumpProfileType
+	// only dump top 10 if set to false, otherwise dump all, only effective when in_text = true
+	DumpFullStack bool
+}
+
+// GetMemOpts return a copy of memOpts
+// if memOpts not exist return a empty memOptions and false
+// memOptions which be returned
+func (o *options) GetMemOpts() (typeOption, bool) {
+	if o.memOpts == nil {
+		return typeOption{}, false
+	}
+	o.memOpts.L.RLock()
+	defer o.memOpts.L.RUnlock()
+	return *o.memOpts, true
+}
+
+// GetCPUOpts return a copy of cpuOpts
+// if cpuOpts not exist return a empty typeOption and false
+func (o *options) GetCPUOpts() (typeOption, bool) {
+	if o.cpuOpts == nil {
+		return typeOption{}, false
+	}
+	o.cpuOpts.L.RLock()
```

Comment:
I'm not sure the lock can protect the consistency that we wanted since the `Set` is not implemented yet?